### DISCUSSION TITLE
如何对方法中的参数进行类型检查，限定参数为任意类型组合中的一种（类型组合例子：【int,double,sting】）

### DISCUSSION THREAD

**Discussion by FLC-GGBOOM** [BYJRK/DotNet-Discussions#388] [Q&A]
此图仅说明我的意图
![1734400176262](https://github.com/user-attachments/assets/19f92145-a4c1-453b-9221-01e4bb780c32)

可以分享您知道的类型检查方法
运行时的类型检查，可以通过拦截器，代码中写类型检查
但是编译前的类型检查应该如何实现

**Comment by BYJRK** [comment]
没有办法。C# 中，面对此类问题，通常的做法是写重载，或者最笨的方式就是写泛型或传入 `object` 类型，然后在运行时进行检查。如果你希望只能传入数字，那么还可以使用 `INumber<>` 接口。

**Reply by FLC-GGBOOM** [reply]
如果方法有多个参数需做类型检查，重载也不好实现，或许还是用运行时检查更合理。感谢解答！

**Reply by BYJRK** [reply]
这种情况的话你或许可以考虑使用 [OneOf](https://github.com/mcintyre321/OneOf)

**Reply by FLC-GGBOOM** [reply]
好的，测试了一下OneOf库可以解决我的问题，非常感谢！
